sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RebaseHelper.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids
```

Comment:
Appears the only reason this isn't in the com.nvidia.spark.rapids package is access to `SparkUpgradeException`.  Would it be better to have a `TrampolineUtil` method to throw this exception and have this method throw directly?  Could also return a `RuntimeException` type if its important to create it but not throw it as well.

Reply:
I wanted to match want spark was doing and throw a `SparkUpgradeException` so it was clear what was happening.  If you want me to move the building of the exception to a trampoline util I can, but I didn't see too much value in that.  Please note that in Spark these same issues exist for Avro, so I made the code as common as possible so we could also support avro with little modification.

Reply:
To me the value of the refactor is to easily identify what package-private Spark methods are being leveraged and reduce our footprint on the `org.apache.spark` package.  It would still be very obvious what's going on even if moved, because the code would be calling `TrampolineUtil.throwSparkUpgradeException("3.0", "reading dates ..`.

However with Catalyst being not-private-yet-effectively-private agree there's not a ton of utility there.  I'm fine with it as-is.